shared/naturalcrit/markdown.js

```diff
@@ -238,7 +268,38 @@ const definitionLists = {
 	}
 };
 
-Marked.use({ extensions: [mustacheSpans, mustacheDivs, mustacheInjectInline, definitionLists] });
+let pageBlockTopLevel = true;
+let pageBlockNumber = 0;
+const pageBlocks = {
+	name  : 'pageBlock',
+	level : 'block',
+	start(src) { return pageBlockTopLevel; },
+	tokenizer(src, tokens) {
+		if(!pageBlockTopLevel) return false;
+		const pageArray = src.split(/^\\page$/gm);
+		if(this.lexer.tokens.length == 0 && this.lexer.state.top) pageBlockNumber = 0;
+		pageBlockNumber++;
+
+		const token = {
+			type       : 'pageBlock',
+			raw        : `${pageArray[0]}\\page`,
+			text       : pageArray[0],
+			pageNumber : pageBlockNumber,
+			tokens     : []
+		};
+
+		pageBlockTopLevel = false;
```

Comment:
I'm not sure what `pageBlockTopLevel` is contributing to this extension?

Reply:
Given that any string - even one that does not contain the `\page` token - should be wrapped in a pageBlock, the processing must then trigger on any given string, including strings that contain other block tokens. But without limitation, this will result in an infinite sequence of pageBlocks.
Given the limitation that no pageBlock can contain another pageBlock, I created the `pageBlockTopLevel` boolean to determine if the current iteration of the tokenizer is running inside a pageBlock token and immediately cancel creating further pageBlock tokens. Perhaps a better name would have been `hasPageBlockParent`, with a default value to `false`.

It's possible that changing the `start(src)` function can be manipulated to produce a similar result.

Reply:
Ok I see. I didn't see the check at the beginning of the Tokenizer.

Yeah I don't think the Start function is doing *anything* now. You can probably just remove it and it should work the same.